### PR TITLE
Add account badge

### PR DESCRIPTION
**Pull request by bumi**
This gives easy access to test if the "user/visitor" has an account, is i
a contributor and if she is core.

Right now I am using this to display an account badge the the name and the account. - currently not styled.

### CODE REVIEW

File: app/services/kredits.js [hunk 1]
```diff
@@ -30,6 +30,17 @@ export default Service.extend({
 
   ethProvider: null,
   currentUserAccounts: null, // default to not having an account. this is the wen web3 is loaded.
+  currentUser: null,
+  currentUserIsContributor: computed('currentUser', function() {
+    return Ember.isPresent(this.get('currentUser'));
```
Comment (fsmanuel) [inline app/services/kredits.js]: You can add:
`import { isEmpty, isPresent } from 'ember-utils';`
and make this `return isPresent(this.get('currentUser'));`

File: app/services/kredits.js [hunk 2]
```diff
@@ -30,6 +30,17 @@ export default Service.extend({
...
+  }),
+  currentUserIsCore: alias('currentUser.isCore'),
+  hasAccounts: computed('currentUserAccounts', function() {
+    return !Ember.isEmpty(this.get('currentUserAccounts'));
```
Comment (fsmanuel) [inline app/services/kredits.js]: ^^

File: app/services/kredits.js [hunk 3]
```diff
@@ -30,6 +30,17 @@ export default Service.extend({
...
+  }),
+  accountNeedsUnlock: computed('currentUserAccounts', function() {
+    return this.get('currentUserAccounts') && Ember.isEmpty(this.get('currentUserAccounts'));
```
Comment (fsmanuel) [inline app/services/kredits.js]: ^^

File: app/services/kredits.js [hunk 4]
```diff
@@ -44,7 +55,14 @@ export default Service.extend({
         ethProvider.listAccounts().then((accounts) => {
           this.set('currentUserAccounts', accounts);
           this.set('ethProvider', ethProvider);
-          resolve(ethProvider);
+          if (accounts.length > 0) {
+            this.get('getCurrentUser').then((contributorData) => {
+              this.set('currentUser', contributorData);
```
Comment (fsmanuel) [inline app/services/kredits.js]: 👍

File: app/services/kredits.js [hunk 5]
```diff
@@ -281,4 +291,23 @@ export default Service.extend({
           });
       });
   },
+
+  getCurrentUser: computed('ethProvider', function() {
+    if (Ember.isEmpty(this.get('currentUserAccounts'))) {
+      return Ember.RSVP.resolve();
```
Comment (fsmanuel) [inline app/services/kredits.js]: We import `RSVP` so you can use it without `Ember`

File: app/services/kredits.js [hunk 6]
```diff
@@ -281,4 +291,23 @@ export default Service.extend({
           });
       });
   },
+
+  getCurrentUser: computed('ethProvider', function() {
+    if (Ember.isEmpty(this.get('currentUserAccounts'))) {
```
Comment (fsmanuel) [inline app/services/kredits.js]: ^^

File: app/services/kredits.js [hunk 7]
```diff
@@ -281,4 +291,23 @@ export default Service.extend({
...
+      return Ember.RSVP.resolve();
+    }
+    return this.get('contributorsContract')
+      .then((contract) => {
+        console.log('user accounts', this.get('currentUserAccounts')[0]);
```
Comment (fsmanuel) [inline app/services/kredits.js]: `let address = this.get('currentUserAccounts.firstObject')`

File: app/services/kredits.js [hunk 8]
```diff
@@ -281,4 +291,23 @@ export default Service.extend({
...
+        return contract.getContributorIdByAddress(this.get('currentUserAccounts')[0])
+          .then((id) => {
+            // check if the user is a contributor or not
```
Comment (fsmanuel) [inline app/services/kredits.js]: `id = id.toNumber()`

File: app/services/kredits.js [hunk 9]
```diff
@@ -281,4 +291,23 @@ export default Service.extend({
...
+            if( id.toNumber() === 0) {
```
Comment (fsmanuel) [inline app/services/kredits.js]: `if (id === 0)`

File: app/templates/index.hbs [hunk 10]
```diff
@@ -1,8 +1,19 @@
+{{#if kredits.hasAccounts }}
+<section id="user-account">
```
Comment (fsmanuel) [inline app/templates/index.hbs]: I would indent it like this:
```hbs
{{#if kredits.hasAccounts }}
  <section id="user-account">
    {{!-- ... --}}
  </section>
{{/if}}
```

**Comment by bumi** [comment]
guess this should get some styling before we merge it.

**Pull request merged**